pkg/app/mutate_cluster_app.go

```diff
@@ -0,0 +1,159 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	releases "github.com/giantswarm/release-operator/v3/api/v1alpha1"
+	"gopkg.in/yaml.v3"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/app-admission-controller/pkg/mutator"
+)
+
+type providerReleaseConfig struct {
+	lastClusterAppVersionWithoutRelease *semver.Version
+}
+
+var providersReleaseConfig = map[string]providerReleaseConfig{
+	"cluster-aws": {
+		lastClusterAppVersionWithoutRelease: semver.New(0, 76, 1, "", ""),
+	},
+}
+
+func (m *Mutator) mutateClusterApp(ctx context.Context, app v1alpha1.App) ([]mutator.PatchOperation, error) {
+	m.logger.Debugf(ctx, "Cluster app mutation for setting App version based on the release. App/Cluster:%s, Namespace:%s\n",
```

Comment:
maybe you should put this log after checking for `isClusterApp`. otherwise it will be always show; which is confusing for non-cluster-apps

Reply:
Sure, makes sense.

Reply:
Done.